src/ima-middleware.js

```diff
@@ -28,11 +28,12 @@ export default class ImaMiddleware extends BaseMiddleware {
 
   /**
    * @constructor
-   * @param {any} context - The plugin context.
+   * @param {Ima} context - The ima plugin context.
    */
-  constructor(context: any) {
+  constructor(context: Ima) {
     super();
-    this._context = context;
+    ctx = context;
+    ctx.player.addEventListener(ctx.player.Event.CHANGE_SOURCE_STARTED, () => this._isPlayerLoaded = false);
```

Comment:
maybe not the best name as player is loaded on change media. maybe change to `isPlayerInitialized` or `isPlayerReady`

Reply:
but the  intention is whether the player is already loaded.
where we are loading it on change media?